sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/expectation/Expectation.scala

```diff
@@ -0,0 +1,236 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2024 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.dataobject.expectation
+
+import com.typesafe.config.Config
+import io.smartdatalake.config.SdlConfigObject.{ActionId, DataObjectId}
+import io.smartdatalake.config._
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.misc.SmartDataLakeLogger
+import io.smartdatalake.util.spark.SparkExpressionUtil
+import io.smartdatalake.workflow.action.ActionHelper
+import io.smartdatalake.workflow.action.generic.transformer.SQLDfTransformer.INPUT_VIEW_NAME
+import io.smartdatalake.workflow.dataframe.spark.SparkColumn
+import io.smartdatalake.workflow.dataframe.{DataFrameFunctions, GenericColumn, GenericDataFrame}
+import io.smartdatalake.workflow.dataobject.ExpectationValidation
+import io.smartdatalake.workflow.dataobject.expectation.ExpectationScope.{ExpectationScope, Job}
+import io.smartdatalake.workflow.dataobject.expectation.ExpectationSeverity.ExpectationSeverity
+import io.smartdatalake.workflow.{ActionPipelineContext, DataFrameSubFeed}
+import org.apache.spark.sql.Column
+
+/**
+ * Trait to define expectations against a dataset.
+ * Expectations are checks based on aggregates over all rows of a dataset.
+ * Through setting the scope of an expectation, it can be controlled if the aggregate is evaluated per job, partition or the whole content of the table.
+ */
+trait BaseExpectation {
+  /**
+   * The name of the expectation
+   */
+  def name: String
+  /**
+   * Optional detailed description of the expectation
+   */
+  def description: Option[String]
+  /**
+   * Severity if expectation fails - can be Error (default) or Warn.
+   * If set to Error, execution will fail, otherwise there will be just a warning logged.
+   */
+  def failedSeverity: ExpectationSeverity
+  /**
+   * The aggregation scope used to evaluate the aggregate expression.
+   * Default is 'Job', which evaluates the records transformed by the current job. This is implemented without big performance impacts on Spark.
+   * Other options are 'All' and 'JobPartition', which are implemented by reading the output data again.
+   */
+  def scope: ExpectationScope = Job
```

Comment:
it's not clear for me what difference the scope has for the end result. An example would be nice somewhere, either in a unit test or here: https://smartdatalake.ch/docs/reference/dataQuality